src/app/txRunner.js

```diff
@@ -69,21 +69,24 @@ TxRunner.prototype.execute = function () {
         if (err) {
           return callback(err, resp)
         }
-
-        if (resp > gasLimit) {
-          return callback('Gas required exceeds limit: ' + resp)
-        }
-
-        tx.gas = resp
-
-        var sendTransaction = self.personalMode ? self.web3.personal.unlockAccountAndSendTransaction : self.web3.eth.sendTransaction
-
-        sendTransaction(tx, function (err, resp) {
+        self.web3.eth.getBlock('latest', function (err, block) {
           if (err) {
-            return callback(err, resp)
-          }
+            return callback(err)
+          } else {
+            var blockGasLimit = Math.floor(block.gasLimit - block.gasLimit / 1024)
+            tx.gas = blockGasLimit < resp ? blockGasLimit : resp
```

Comment:
I'm just wondering, shouldn't we just take it as an error if the estimate is higher than the block limit? It will very likely fail.